### ISSUE TITLE
FB8552424: AVAudioRecorder should be more resilient to corruptions

### ISSUE THREAD

**Issue by sindresorhus** [feedback-assistant/reports#146]
- **Date:** 2020-08-27
- **Resolution:** Open
- **Area:** AVFoundation
- **OS:** macOS 10.15.6
- **Type:** Suggestion

## Description

Currently, if my app crashes (exception, SIGTERM, etc) while AVAudioRecorder is recording (`.isRecording == true`), the resulting file becomes corrupt. Reproduced when using the format `kAudioFormatMPEG4AAC`. I think AVAudioRecorder should do its utter best to prevent corruption. App/computer crashes happen and the user’s data is precious.

In my app, I’ve added a SIGTERM signal handler which calls `AVAudioRecorder#stop()`. But I think AVAudioRecorder should do this by default. There are also other crashes which I cannot handle in Swift, like an Objective-C exception.

So in short: AVAudioRecorder should call `.stop()` on the recorder before exiting the app no matter what (as far as it’s possible).
